### PR TITLE
Feature - S3 / R2 Backend

### PR DESCRIPTION
**Pull request by jack828**
## What it does

S3 compatibility - AWS and R2 are tested and supported

## Tests Written?

 - [x] Yes, I've been a good codeperson and written tests.

## Docs updated

 - [x] Yes, I've been a good codeperson and updated the docs.

## Comments

Might break existing mongo gridfs, but i doubt it as the tests pass

## Changelog Snippet

Already updated!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,7 +35,15 @@ module.exports =
...
 , key: 'key'
+, backend: 'mongo'
 , databaseUri: 'mongodb://localhost:27017/darkroom'
+
+// Or use S3
+, backend: 'S3'
+, accessKeyId: 'FOOBARCAFEBABE'
+, secretAccessKey: 'opefkofwpokfeaioajfew/efwkoewkfopw'
+, region: 'eu-west-2'
+, bucket: 'darkroom'
```
Comment (Asheboy) [inline README.md:46]: are these real?

Comment (jack828) [reply README.md:46]: Absolutely not

Comment (jack828) [reply README.md:46]: Can you imagine how silly you need to be to push legit aws secrets to a public GitHub repo?

Comment (jack828) [reply README.md:46]: Answer: this silly https://github.com/clocklimited/Darkroom-api/compare/6fda036e903a077a2ffba2ddf581e13d3f125f51..136e1a756c379198330d02a9d03144bf11ba2a47

**Pull request merged**